.github/workflows/python-poetry-publish.yaml

```diff
@@ -0,0 +1,53 @@
+name: Reusable workflow for publishing to PyPI
+
+on:
+  workflow_call:
+    inputs:
+      publish-to-test:
+        description: "If set to false, the packages are published to PyPI. (Default is true)"
+        required: false
+        default: "true"
+        type: string
+      python-version:
+        description: "The Python version for the Poetry virtual environment. (Default is 3.10)"
+        required: false
+        default: "3.10"
+        type: string
+      poetry-version:
+        description: "The Poetry version to be installed. (Default is 1.5.1)"
+        required: false
+        default: "1.5.1"
+        type: string
+      working-directory:
+        description: "The working directory of your Python packages. (Default is root directory)"
+        required: false
+        default: "./"
+        type: string
+      ref:
+        description: "The ref name to checkout the repository."
+        required: false
+        default: ${{ github.event.repository.default_branch }}
```

Comment:
Shouldn't the default be the current branch in case we want to build and publish on a feature or tag branch?

Reply:
Good point. What do you think @philipp94831 
@disrupted

Reply:
yeah, makes sense. then we wouldn't even need the input at all. The checkout action already uses the current ref by default

Reply:
I have removed the `ref` input now